feature-libs/asm/components/customer-list/customer-list.component.ts

```diff
@@ -0,0 +1,243 @@
+  ICON_TYPE,
+  ModalService,
+} from '@spartacus/storefront';
+import { combineLatest, Observable, of } from 'rxjs';
+import { filter, map, switchMap, tap } from 'rxjs/operators';
+import { CustomerListActionEvent } from './customer-list.model';
+
+@Component({
+  selector: 'cx-customer-list',
+  templateUrl: './customer-list.component.html',
+})
+export class CustomerListComponent implements OnInit {
+  private PAGE_SIZE = 5;
+
+  focusConfig: FocusConfig = {
+    trap: true,
+    block: true,
+    autofocus: 'customer-list-selector',
+    focusOnEscape: true,
+  };
+
+  iconTypes = ICON_TYPE;
+
+  BREAKPOINT = BREAKPOINT;
+
+  selectedUserGroupId: string | undefined;
+
+  customerSearchPage$: Observable<CustomerSearchPage | null>;
+
+  customerListsPage$: Observable<CustomerListsPage | undefined>;
+
+  selectedCustomer: User;
+
+  currentPage = 0;
+
+  maxPage = 0;
+
+  loaded = false;
+
+  sorts: SortModel[] | null;
+
+  sortCode: string | undefined;
+
+  breakpoint$: Observable<BREAKPOINT>;
+
+  customerListConfig = this.asmConfig?.asm?.customerList;
+
+  constructor(
+    protected modalService: ModalService,
+    protected asmService: AsmService,
+    protected breakpointService: BreakpointService,
+    protected asmConfig: AsmConfig,
+    protected translation: TranslationService
+  ) {
+    this.breakpoint$ = this.getBreakpoint();
+  }
+
+  ngOnInit(): void {
+    this.customerListsPage$ = this.asmService
+      .getCustomerLists()
+      .pipe(
+        filter((queryState) => queryState.loading === false),
+        map((queryState) => {
+          return queryState.data;
+        })
+      )
+      .pipe(
+        map((customerListsPage) => customerListsPage),
+        filter(() => true),
+        // set the first value of this.customerListsPage$ to be selected
+        tap((result) => {
+          if (!this.selectedUserGroupId) {
+            this.selectedUserGroupId = result?.userGroups?.[0].uid;
+            this.sorts = null;
+            this.fetchCustomers();
+          }
+        })
+      );
+  }
+
+  fetchCustomers(): void {
+    const pageSize =
+      this.asmConfig.asm?.customerList?.pageSize ?? this.PAGE_SIZE;
+    if (this.selectedUserGroupId) {
+      const options: CustomerSearchOptions = {
+        customerListId: this.selectedUserGroupId,
+        pageSize: pageSize,
+        currentPage: this.currentPage,
+      };
+      if (this.sortCode) {
+        options.sort = this.sortCode;
+      }
+      this.customerSearchPage$ = of(options)
+        .pipe(
+          tap(() => (this.loaded = false)),
+          switchMap((options) => this.asmService.searchCustomers(options))
+        )
+        .pipe(
+          tap((result) => {
+            this.loaded = true;
+            if (result.sorts) {
+              this.sorts = result.sorts;
+              this.sortCode = result.pagination?.sort;
+            }
+            if (result.entries.length < pageSize) {
+              this.maxPage = result.pagination?.currentPage || 0;
```

Comment:
`maxPage` here is confusing. Maybe replaced by a boolean variable `isLastPage`. (`this.isLastPage = result.entries.length < pageSize`)